From 39477907c3ddb3692b9305c2eb36683ade9df806 Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Wed, 1 Jul 2020 22:32:21 +0530
Subject: [PATCH 1416/1852] spi: spi-zynqmp-gqspi: Update correct value for
 DLY1 bit field

commit 767ba34da2cafa553c8b1042d52234d56fec791e from
https://github.com/Xilinx/linux-xlnx.git

The driver was using an incorrect value for DLY1 bit field in the GQSPI
loopback clock delay adjustment register. This bit gets set in Versal
when the qspi operating frequency is greater than 100Mhz and the mode
was not validated earlier. Fixed it by updating DLY1 bit field in the
GQSPI loopback clock delay adjustment register with the correct value
as recommended in the IP spec.

Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Acked-by: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/spi/spi-zynqmp-gqspi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-zynqmp-gqspi.c b/drivers/spi/spi-zynqmp-gqspi.c
index 5fc514597f78..2652d8ffb423 100644
--- a/drivers/spi/spi-zynqmp-gqspi.c
+++ b/drivers/spi/spi-zynqmp-gqspi.c
@@ -142,7 +142,7 @@
 #define GQSPI_TX_BUS_WIDTH_DUAL		0x2
 #define GQSPI_TX_BUS_WIDTH_SINGLE	0x1
 #define GQSPI_LPBK_DLY_ADJ_LPBK_SHIFT	5
-#define GQSPI_LPBK_DLY_ADJ_DLY_1	0x2
+#define GQSPI_LPBK_DLY_ADJ_DLY_1	0x1
 #define GQSPI_LPBK_DLY_ADJ_DLY_1_SHIFT	3
 #define GQSPI_LPBK_DLY_ADJ_DLY_0	0x3
 #define GQSPI_USE_DATA_DLY		0x1
-- 
2.31.1

